config/apigroup_config.go

```diff
@@ -0,0 +1,714 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+// This file is generated
+// nolint:misspell
+var (
+	apiGroupMap = map[string]string{
```

Comment:
It turns out Terraform native Azure provider [does not have](https://github.com/hashicorp/terraform-provider-azurerm/blob/baf55926fe813011003ee4fb0e8e6134fcfcca87/internal/provider/provider.go#L92) a flat structure like the native AWS provider. It uses more structure (e.g., interface implementations) for registering resources. Thus, it won't be as straightforward as the AWS version to implement a scheme based on the Terraform package names, though it should still be doable. 

But as a defaulting mechanism I also liked the idea of relying on the grouping chosen by Microsoft as discussed [here](https://github.com/crossplane-contrib/provider-jet-azure/pull/118#discussion_r778595791) because of low maintenance cost and high consistency with the native (REST) implementation. So we can keep these native provider names as the defaults and provide any manual override configuration under `config/<package>`.

Reply:
Thank you @muvaf for the clarification, I think I had taken it in the broader context. I've implemented a mechanism that defaults to dropping the first (`_`-separated) token and constructing the Kind name from the rest, and that allows overriding the number of initial tokens to be dropped for the entire group. And for cases where I would like to override only a specific Kind name (and not the Kind names in the entire group), I have provided custom configurations under the `config/<group>` folder. The logic is defaults affecting to multiple resources are specified in a common place, but any specific overrides reside under `config/<group>` together with any future other configuration overrides like references, etc.

I think this logical organization makes sense but I can also copy the implementation in jet aws & gcp and maybe move those specific overrides to a common place like it's done there. It looks more like our initial approach of putting all configuration into a single package.

Reply:
You can use the same implementation but it seems like the one you added does the job pretty well already, so all should be fine. Thank you!